tests/parametric/test_headers_tracecontext.py

```diff
@@ -807,6 +807,165 @@ def test_tracestate_w3c_p_inject(self, test_agent, test_library):
             # FIXME: nodejs paramerric app sets span.span_id to a string, convert this to an int
             assert "p:{:016x}".format(int(span.span_id)) in tracestate
 
+    # W3C Phase 3 to try adding the tag if the span id matches regardless of headers order(if tracecontext is accounted)
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "cpp", reason="Not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_PROPAGATION_STYLE": "datadog,tracecontext"}])
+    def test_tracestate_w3c_p_phase_3_extract_inject(self, test_agent, test_library):
+        """
+        Ensure the last parent id tag is set according to the W3C phase 3 spec
+        """
+        with test_library:
+
+            # 1) Datadog and tracecontext headers, trace ids and parent ids match
+            with test_library.start_span(
+                name="matching_trace_id_and_span_id",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000001-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-trace-id", "1"],
+                    ["x-datadog-parent-id", "987654321"],
+                ],
+            ):
+                pass
+
+            # 2) Datadog and tracecontext headers, trace ids and parent ids match, missing p
```

Comment:
`trace ids and parent ids match`

I don't think the trace ids match here, `traceparent` has defined a valid 128-bit trace ID, but we only have the lower 64-bit trace ID in the Datadog headers.

Reply:
The RFC sais that id the Datadog headers define a 64-bit trace id (as it does here, because `x-datadog-tags[tid]` is missing), we should ignore the upper 64 bits.

Fwiw, I question that decision to ignore the upper 64 bits. I think we should only ignore them if they are all zeros.

Reply:
**NOTE:** RFC  has been updated with very detailed expectations.